src/test/ui/lint/rfc-2383-lint-reason/expect_unfulfilled_expectation.rs

```diff
@@ -0,0 +1,39 @@
+// check-pass
+// ignore-tidy-linelength
+
+#![feature(lint_reasons)]
+#![warn(unused_mut)]
+
+#![expect(unfulfilled_lint_expectations, reason = "idk why you would expect this")]
+//~^ WARNING this lint expectation is unfulfilled
+//~| NOTE `#[warn(unfulfilled_lint_expectations)]` on by default
+//~| NOTE idk why you would expect this
+//~| NOTE the `unfulfilled_lint_expectations` lint can't be expected and will always produce this message
+
+#[expect(unfulfilled_lint_expectations, reason = "a local: idk why you would expect this")]
+//~^ WARNING this lint expectation is unfulfilled
+//~| NOTE a local: idk why you would expect this
+//~| NOTE the `unfulfilled_lint_expectations` lint can't be expected and will always produce this message
+pub fn normal_test_fn() {
+    #[expect(unused_mut, reason = "this expectation will create a diagnostic with the default lint level")]
+    //~^ WARNING this lint expectation is unfulfilled
+    //~| NOTE this expectation will create a diagnostic with the default lint level
+    let mut v = vec![1, 1, 2, 3, 5];
+    v.sort();
+
+    // Check that lint lists including `unfulfilled_lint_expectations` are also handled correctly
+    #[expect(unused, unfulfilled_lint_expectations, reason = "the expectation for `unused` should be fulfilled")]
+    //~^ WARNING this lint expectation is unfulfilled
+    //~| NOTE the expectation for `unused` should be fulfilled
+    //~| NOTE the `unfulfilled_lint_expectations` lint can't be expected and will always produce this message
+    let value = "I'm unused";
+}
+
+#[expect(warnings, reason = "this suppresses all warnings and also suppresses itself. No warning will be issued")]
+pub fn expect_warnings() {
+    // This lint trigger will be suppressed
+    #[warn(unused_mut)]
```

Comment:
I just wondered why this doesn't warn here. It should override the`expect`/`allow` of the outer scope. But it seems like `warnings` is somehow treated differently to other lint groups. It behaves the same as with `allow`, see [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=205e429f0911b448c904b773ee5c2785), so this is not a bug in the `expect` attribute.

But I would argue, that this is generally surprising behavior. Is this documented somewhere?

Reply:
Yes, the `warnings` name seems to behave a bit differently. It looks like it sets the level for all warnings in a given scope. I only could find this documentation: 

![image](https://user-images.githubusercontent.com/17087237/157296495-e4394d3c-b93f-417f-a8d3-9bc67d630698.png)

[source](https://doc.rust-lang.org/rustc/lints/groups.html) and [this](https://rust-unofficial.github.io/patterns/anti_patterns/deny-warnings.html). But both say that it behaves like a lint group and not overrides the level. I think this is fine for now as it behaves like the `allow` attribute :upside_down_face:

Reply:
> I think this is fine for now

Agreed. Nothing to do in this PR. I just wanted to point out this behavior, because I thought it is _generally_ unexpected.